siliconcompiler/core.py

```diff
@@ -1530,6 +1530,31 @@ def _check_files(self):
 
         return True
 
+    ###########################################################################
+    def check_filepaths(self):
+        '''
+        Verifies that paths to all files in manifest are valid.
+        '''
+
+        allkeys = self.getkeys()
+        for keypath in allkeys:
+            allpaths = []
+            paramtype = self.get(*keypath, field='type')
+            if 'file' in paramtype or 'dir' in paramtype:
+                if 'dir' not in keypath and self.get(*keypath):
+                    allpaths = list(self.get(*keypath))
+                for path in allpaths:
+                    #check for env var
+                    m = re.match(r'\$(\w+)(.*)', path)
```

Comment:
Could we use the `_resolve_env_vars` method for this, or is the logic a bit different?

Reply:
Possibly...